File: src/main/java/pl/training/workshop/m5/s14_reuse/step1/CorporateAccount.java
package pl.training.workshop.m5.s14_reuse.step1;

/**
 * Krok 1: bez zmian - nadal dziedziczy (i nadal łamie zastępowalność).
 */
public class CorporateAccount extends LoyaltyAccount {
    public CorporateAccount(String company) {
        super(company);
    }

    @Override
    public boolean redeemFreeTicket() {
        throw new UnsupportedOperationException("konto firmowe nie wymienia punktów na bilety");
    }
}

File: csharp/src/Training.Workshop/M5/S14Reuse/Step1/CorporateAccount.cs
namespace Training.Workshop.M5.S14Reuse.Step1;

/// <summary>Krok 1: bez zmian - nadal dziedziczy (i nadal łamie zastępowalność).</summary>
public class CorporateAccount : LoyaltyAccount
{
    public CorporateAccount(string company) : base(company)
    {
    }

    public override bool RedeemFreeTicket()
    {
        throw new NotSupportedException("konto firmowe nie wymienia punktów na bilety");
    }
}
